### PR TITLE
HMC ack support added for LogEntry

### PR DESCRIPTION
**Pull request by vflobo**
A new property is added to support hmc ack

### CODE REVIEW

File: redfish-core/lib/log_services.hpp [hunk 1]
```diff
@@ -1795,6 +1830,13 @@ inline void requestRoutesDBusEventLogEntry(App& app)
                                 filePath = std::get_if<std::string>(
                                     &propertyMap.second);
                             }
+
+#ifdef BMCWEB_ENABLE_IBM_MANAGEMENT_CONSOLE
+                            else if (propertyMap.first == "ManagementSystemAck")
+                            {
+                                bool* managementsaptr =
+                                    std::get_if<bool>(&propertyMap.second);
+                                if (managementsaptr == nullptr)
```
Comment (gtmills) [inline redfish-core/lib/log_services.hpp:1839]: Looks like a bad merge resolve?

Comment (vflobo) [reply redfish-core/lib/log_services.hpp:1839]: yes, I am going to fix and test it on Hw and then upload again .

**Comment by sunharis** [comment]
Is this commit up-streamed ?

**Comment by gtmills** [comment]
Did the validator pass ?

> Is this commit up-streamed ?

I think we just decided this should be downstream only, since we haven't tried to add to Redfish. @spinler That should right to you?

**Comment by vflobo** [comment]
> 
> 
> Did the validator pass ?
> Yes
> > Is this commit up-streamed ?
> 
> I think we just decided this should be downstream only, since we haven't tried to add to Redfish. @spinler That should right to you?

**Comment by spinler** [comment]
> Did the validator pass ?
> 
> > Is this commit up-streamed ?
> 
> I think we just decided this should be downstream only, since we haven't tried to add to Redfish. @spinler That should right to you?

Right, downstream only because we are using org.open_power.PEL interfaces.

File: redfish-core/lib/log_services.hpp [hunk 2]
```diff
@@ -1880,31 +1906,72 @@ inline void requestRoutesDBusCELogEntryCollection(App& app)
...
+        {
+            BMCWEB_LOG_ERROR << "Could not read managementSystemAck";
+            return;
+        }
+    }
+    if (managementSystemAck.has_value())
+    {
+        BMCWEB_LOG_DEBUG << "Set ManagementSystemAck";
+        crow::connections::systemBus->async_method_call(
+            [asyncResp, entryId](const boost::system::error_code ec) {
```
Comment (patelabhishek9893) [inline redfish-core/lib/log_services.hpp:1960]: `entryId` does not need to pass inside captures of lambda

**Review by patelabhishek9893 — CHANGES_REQUESTED** [review]

File: redfish-core/lib/log_services.hpp (hunk 2 above)
Comment (vflobo) [reply redfish-core/lib/log_services.hpp]: Done

File: redfish-core/lib/log_services.hpp [hunk 3]
```diff
@@ -1880,31 +1906,72 @@ inline void requestRoutesDBusCELogEntryCollection(App& app)
...
-        "xyz.openbmc_project.Logging",
-        "/xyz/openbmc_project/logging/entry/" + entryId,
-        "org.freedesktop.DBus.Properties", "Set",
-        "xyz.openbmc_project.Logging.Entry", "Resolved",
-        std::variant<bool>(*resolved));
+    if (resolved.has_value())
+    {
+        BMCWEB_LOG_DEBUG << "Set Resolved";
+        crow::connections::systemBus->async_method_call(
+            [asyncResp, entryId](const boost::system::error_code ec) {
```
[the remark this answers is not in the document]
Comment (vflobo) [reply redfish-core/lib/log_services.hpp]: done

File: redfish-core/lib/log_services.hpp [hunk 4]
```diff
@@ -1724,8 +1727,25 @@ inline void getDBusLogEntryCollection(
                             messages::internalError(asyncResp->res);
                             return;
                         }
-                        break;
                     }
+#ifdef BMCWEB_ENABLE_IBM_MANAGEMENT_CONSOLE
+                    else if (propertyMap.first == "ManagementSystemAck")
+                    {
+                        std::cerr << "--IN ManagementSystemAck"
```
[the remark this answers is not in the document]
Comment (vflobo) [reply redfish-core/lib/log_services.hpp]: Done

File: static/redfish/v1/JsonSchemas/OemLogEntry/index.json [hunk 5]
```diff
@@ -0,0 +1,42 @@
+{
+    "$id": "http://redfish.dmtf.org/schemas/v1/OemLogEntry.v1_0_0.json",
+    "$schema": "http://redfish.dmtf.org/schemas/v1/redfish-schema-v1.json",
+    "copyright": "Copyright 2014-2019 DMTF. For the full DMTF copyright policy, see http://www.dmtf.org/about/policies/copyright",
```
Comment (gtmills) [inline static/redfish/v1/JsonSchemas/OemLogEntry/index.json:4]: A copyright of DMTF, doesn't sound right. 
Oh well.

**Review by gtmills — APPROVED** [review]
Just want to make sure the validator passes

File: redfish-core/lib/log_services.hpp [hunk 6]
```diff
@@ -2072,6 +2150,12 @@ inline void getDBusLogEntry(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
     asyncResp->res.jsonValue["Created"] = crow::utility::getDateTime(timestamp);
     asyncResp->res.jsonValue["Modified"] =
         crow::utility::getDateTime(updateTimestamp);
+#ifdef BMCWEB_ENABLE_IBM_MANAGEMENT_CONSOLE
+    asyncResp->res.jsonValue["Oem"]["OpenBMC"]["@odata.type"] =
+        "#OemLogEntry.v1_0_0.LogEntry";
```
[the remark this answers is not in the document]
Comment (vflobo) [reply redfish-core/lib/log_services.hpp:2155]: {
    "@odata.id": "/redfish/v1/SessionService/Sessions/24Wu8TAT5x",
    "@odata.type": "#Session.v1_3_0.Session",
    "ClientOriginIPAddress": "::ffff:9.41.165.94",
    "Description": "Manager User Session",
    "Id": "24Wu8TAT5x",
    "Name": "User Session",
    "Oem": {
        "OpenBMC": {
            "@odata.type": "#OemSession.v1_0_0.Session",
            "ClientID": ""
        }
    },
    "UserName": "root"
}

The above one is from sessions OEM property and it has the same format which I used as an example.
The Validator passes

**Comment by vflobo** [comment]
> 
> 
> Just want to make sure the validator passes

Yes

**Comment by vflobo** [comment]
> 
> 
> Validator passed?

Yes

**Review by spinler — APPROVED** [review]

**Pull request merged**